numpy/distutils/system_info.py

```diff
@@ -498,17 +505,46 @@ def calc_libraries_info(self):
                 dict_append(info, **i)
             else:
                 log.info('Library %s was not found. Ignoring' % (lib))
+            i = self.check_libs(r_dirs, [lib])
+            if i is not None:
+                # Swap library keywords found to runtime_library_dirs
+                # the libraries are insisting on the user having defined
+                # them using the library_dirs, and not necessarily by
+                # runtime_library_dirs
+                del i['libraries']
+                i['runtime_library_dirs'] = i.pop('library_dirs')
+                dict_append(info, **i)
```

Comment:
The above looks correct, but it took me a couple of minutes to understand the comment and the use of ``del``. How about replacing the whole if clause with:

    # From `check_libs` we get keys "libraries" and "library_dirs",
    # while we need only key "runtime_library_dirs" with those paths
    dict_append(info, runtime_library_dirs=i['library_dirs'])